### PR TITLE
Feature/9389 create technical behavior for qr scanner

### PR DESCRIPTION
**Pull request by 30mar**
## Description
Create main logic for the general QR code to handle different QRCodes and return the correct model 

## Link to Jira
https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-9389

### CODE REVIEW

File: src/xcode/ENA/ENA/Source/Scenes/QRScanner/CoronaTestsParser.swift [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+	
+	// MARK: - Init
+
+	init() {
+
+//		#if DEBUG
+//		if isUITesting {
+//			onSuccess(CoronaTestRegistrationInformation.pcr(guid: "guid"))
+//		}
+//		#endif
```
Comment (pascal-brause) [inline src/xcode/ENA/ENA/Source/Scenes/QRScanner/CoronaTestsParser.swift:19]: what about this?

File: src/xcode/ENA/ENA/Source/Scenes/QRScanner/QRScannerViewModel.swift [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+
+	private func parseTheExtractedQRCode(url: String) {
+		// Check the prefix to know which type
+		// if we go directly and try to parse we might get an incorrect error
+		// e.g: scanning a PCR QRCode and trying to parse it at a health-certificate, we will get a healthCertificate related error
+		// which is incorrect and it should be a Corona test error, so we need to have an idea about the type of qrcode before paring it
+		
+		let traceLocationsPrefix = "e.coronawarn.app"
+		let antigetTestPrefix = "https://s.coronawarn.app"
+		let pcrPrefix = "https://localhost"
```
Comment (pascal-brause) [inline src/xcode/ENA/ENA/Source/Scenes/QRScanner/QRScannerViewModel.swift:162]: sure here?

File: src/xcode/ENA/ENA/Source/Scenes/QRScanner/QRScannerViewModel.swift [hunk 3]
```diff
@@ -0,0 +1,190 @@
+//
+// 🦠 Corona-Warn-App
+//
+
+import AVFoundation
+
+protocol QRCodeParsable {
+	func parse(qrCode: String, completion: @escaping (Result<QRCodeResult, QRCodeParserError>) -> Void)
```
Comment (pascal-brause) [inline src/xcode/ENA/ENA/Source/Scenes/QRScanner/QRScannerViewModel.swift:8]: Small comment would be fine

File: src/xcode/ENA/ENA/Source/Scenes/QRScanner/QRScannerViewModel.swift [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+			}
+
+		} catch {
+			Log.error(error.localizedDescription, log: .api)
+		}
+	}
+
+	var isScanningActivated: Bool {
+		captureSession?.isRunning ?? false
+	}
```
Comment (pascal-brause) [inline src/xcode/ENA/ENA/Source/Scenes/QRScanner/QRScannerViewModel.swift:130]: sort it on top of internal things.

File: src/xcode/ENA/ENA/Source/Scenes/QRScanner/QRScannerViewModel.swift [hunk 5]
```diff
@@ -0,0 +1,190 @@
...
+		parser.parse(qrCode: url, completion: { result in
+			self.completion(result)
+		})
+	}
+	
+	private let captureDevice: AVCaptureDevice?
+	private var parser: QRCodeParsable?
+	private let verificationHelper: QRCodeVerificationHelper
+	private let appConfiguration: AppConfigurationProviding
+	private let healthCertificateService: HealthCertificateService
```
Comment (pascal-brause) [inline src/xcode/ENA/ENA/Source/Scenes/QRScanner/QRScannerViewModel.swift:189]: sort accordingly.
And then put them above func's.

File: src/xcode/ENA/ENA/Source/Scenes/QRScanner/QRScannerViewModel.swift (hunk 2 above)
Comment (pascal-brause) [reply src/xcode/ENA/ENA/Source/Scenes/QRScanner/QRScannerViewModel.swift:162]: seems so.

**Review by naveeddotio — APPROVED** [review]

**Review by pascal-brause — APPROVED** [review]

**Pull request merged**